app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -142,102 +132,33 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
-        size_bottom_spacer();
-        listView.addFooterView(this.bottom_list_spacer);
-
-        listView.setAdapter(adapter);
-
-        // We want blue highlights when you longpress
-        listView.setDrawSelectorOnTop(true);
-
-        registerForContextMenu(listView);
-
-        listView.setOnScrollListener(new OnScrollListener() {
-
+        recyclerView = (RecyclerView) view.findViewById(R.id.recyclerView);
+        linearLayoutManager = new LinearLayoutManager(getContext());
+        recyclerView.setLayoutManager(linearLayoutManager);
+        adapter = new RecyclerMessageAdapter(messageList, getActivity(), (filter==null));
+        recyclerView.addItemDecoration(new HeaderSpaceItemDecoration(24));
+        recyclerView.setAdapter(adapter);
+        registerForContextMenu(recyclerView);
+        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
             @Override
-            public void onScroll(AbsListView view, int firstVisibleItem,
-                                 int visibleItemCount, int totalItemCount) {
-
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
                 final int near = 6;
-
                 if (!paused && !loadingMessages && firstMessageId > 0
                         && lastMessageId > 0) {
-                    if (firstVisibleItem + visibleItemCount > totalItemCount
-                            - near && !loadedToBottom) {
+                    if (linearLayoutManager.findLastVisibleItemPosition() == adapter.getItemCount() - 1) {
                         // At the bottom of the list
                         Log.i("scroll", "Starting request below");
                         loadMoreMessages(LoadPosition.BELOW);
                     }
-                    if (firstVisibleItem < near && !loadedToTop) {
+                    if (linearLayoutManager.findFirstCompletelyVisibleItemPosition() < near && !loadedToTop) {
                         // At the top of the list
                         Log.i("scroll", "Starting request above");
                         loadMoreMessages(LoadPosition.ABOVE);
                     }
                 }
-
-            }
-
-            @Override
-            public void onScrollStateChanged(AbsListView view, int scrollState) {
-                try {
-                    // Scrolling messages isn't meaningful unless we have
-                    // messages to scroll.
-                    Message message = (Message) view.getItemAtPosition(view
-                            .getFirstVisiblePosition());
-                    int mID = message.getID();
-                    if (filter == null && app.getPointer() < mID) {
-                        Log.i("scrolling", "Now at " + mID);
-                        (new AsyncPointerUpdate(app)).execute(mID);
-                        app.setPointer(mID);
-                    }
-
-                    app.markMessageAsRead(message);
```

Comment:
It looks like you've removed the functionality that marks messages as read on scroll.  This is probably behavior worth preserving.